src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+        },
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        renderEditSidebar: {
+            func: "{that}.renderMarkup",
+            args: ["editWidgetSidebar", "{that}.options.templates.editPrefWidget", "{arguments}.0"]
+        },
+        editValueEvent: {
+            funcName: "gpii.devpmt.editValueEvent",
+            args: ["{that}", "{arguments}.0"]
+        },
+        editProductEnabled: {
+            funcName: "gpii.devpmt.editProductEnabled",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+        },
+        savePrefset: {
+            funcName: "gpii.devpmt.savePrefset",
+            args: ["{that}" /*, "{arguments}.0" */]
+        },
+        updateMetadataFromPrefs: {
+            funcName: "gpii.devpmt.updateMetadataFromPrefs",
+            args: ["{that}"]
+        },
+        initSettingTableWidgets: {
+            funcName: "gpii.devpmt.initSettingTableWidgets",
+            args: ["{that}"]
+        },
+        openAddProductDialog: {
+            funcName: "gpii.devpmt.openAddProductDialog",
+            args: ["{that}", "{arguments}.0"] // appId
+        },
+        toggleDevModeView: {
+            funcName: "gpii.devpmt.toggleDevModeView",
+            args: ["{that}", "{that}.model.devModeOn"]
+        },
+        lookupGenericPrefValue: {
+            funcName: "gpii.devpmt.lookupGenericPrefValue",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"] // context, commonTerm
+        },
+        lookupProductPrefValue: {
+            funcName: "gpii.devpmt.lookupProductPrefValue",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // context, product, settingTerm
+        },
+        onDeleteContext: {
+            funcName: "gpii.devpmt.onDeleteContext",
+            args: ["{that}", "{arguments}.0"] // event
+        }
+    },
+    listeners: {
+        "onCreate": {
+            funcName: "gpii.devpmt.npsetInit",
+            args: ["{that}"] //"onCreate listener"]
+        },
+        "onMarkupRendered": [
+            {
+                "this": "{that}.dom.topbarSaveButton",
+                "method": "click",
+                args: ["{that}.events.openConfirmSaveDialog.fire"]
+            },
+            {
+                "this": "{that}.dom.deleteContextButtons",
+                "method": "click",
+                args: ["{that}.onDeleteContext"]
+            },
+            {
+                "this": "{that}.dom.devModeIcon",
+                "method": "click",
+                args: ["{that}.toggleDevModeView"]
+            },
+            {
+                funcName: "gpii.devpmt.updateFoundationSticky",
+                args: []
+            },
+            {
+                func: "{that}.initSettingTableWidgets"
+            }
+        ]
+    },
+    modelListeners: {
+        "flatPrefs": [{
```

Comment:
You probably want to namespace these and then sequence them using priority definitions. https://docs.fluidproject.org/infusion/development/Priorities.html

Reply:
this is done